apis/object/v1alpha1/types.go

```diff
@@ -74,3 +127,46 @@ type ObjectList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Object `json:"items"`
 }
+
+// ApplyFromFieldPathPatch patches the "to" resource, using a source field
+// on the "from" resource.
+func (r *Reference) ApplyFromFieldPathPatch(from, to runtime.Object) error {
+	if r.FromObject.FieldPath == nil {
+		return nil
+	}
+
+	// Default to patch the same field on the "to" resource.
+	if r.ToFieldPath == nil {
+		r.ToFieldPath = r.FromObject.FieldPath
+	}
+
+	fromMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(from)
+	if err != nil {
+		return err
+	}
+
+	out, err := fieldpath.Pave(fromMap).GetValue(*r.FromObject.FieldPath)
+	if err != nil {
+		return err
+	}
+
+	return patchFieldValueToObject(*r.ToFieldPath, out, to)
+}
+
+// patchFieldValueToObject, given a path, value and "to" object, will
+// apply the value to the "to" object at the given path, returning
+// any errors as they occur.
+func patchFieldValueToObject(path string, value interface{}, to runtime.Object) error {
+	if u, ok := to.(interface{ UnstructuredContent() map[string]interface{} }); ok {
+		return fieldpath.Pave(u.UnstructuredContent()).SetValue(path, value)
+	}
+
+	toMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(to)
+	if err != nil {
+		return err
+	}
+	if err := fieldpath.Pave(toMap).SetValue(path, value); err != nil {
+		return err
+	}
+	return runtime.DefaultUnstructuredConverter.FromUnstructured(toMap, to)
```

Comment:
Seems like we can just use existing fieldpath functions, if I am not missing something.

```suggestion
	paved, err := fieldpath.PaveObject(to)
	if err != nil {
		return err
	}
	return paved.SetValue(path, value)
```

Reply:
I found that I still need `runtime.DefaultUnstructuredConverter.FromUnstructured` at last, as I need to "dump" the modified contents from `paved` to `to`, otherwise the changes will be lost.